### PR TITLE
Fix annotating CI run when product tests fail

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

The product tests job test step should have an ID used to check if it failed, so its test reports can be processed and annotations can be generated.

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1045,6 +1045,7 @@ jobs:
           !contains(github.event.pull_request.labels.*.name, 'tests:all-product')
         run: echo "PTL_OPTS=--impacted-features impacted-features.log" >> $GITHUB_ENV
       - name: Product Tests
+        id: tests
```
Comment (raunaqmorarka) [inline .github/workflows/ci.yml:1048]: product-tests ?

**Review by raunaqmorarka — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nineinchnick) [reply .github/workflows/ci.yml:1048]: It's not important to distinguish the kind of tests here, and it's better to keep it consistent across all jobs. Very few jobs have more than one step that runs tests, and it has a suffix then.

**Review by wendigo — APPROVED** [review]

**Comment by wendigo** [comment]
@nineinchnick there is no annotation here

**Comment by nineinchnick** [comment]
I think this is because the environment failed to start, and it didn't even run any tests. If `tempto` was not started, it couldn't have generated any `testng-report.xml` file:
```
2024-09-19T10:17:41.626Z	INFO	main	io.trino.tests.product.launcher.env.Environment	Environment 'multinode-sqlserver' failed to start in attempt(s): 6: java.lang.RuntimeException: java.util.concurrent.ExecutionException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image mcr.microsoft.com/mssql/server:2017-latest
2024-09-19T10:17:41.626Z	WARN	main	io.trino.tests.product.launcher.cli.TestRun	Failed to execute tests
java.lang.RuntimeException: java.util.concurrent.ExecutionException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image mcr.microsoft.com/mssql/server:2017-latest
```

**Pull request merged**